scripts/generate_psa_key_attributes.py

```diff
@@ -0,0 +1,238 @@
+        return False
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Generate PSA key attributes and write to stdout."
+        "Use 'xxd' or similar if you want to print it to a terminal:"
+        f" 'python {Path(__file__).name} --args | xxd'",
+        allow_abbrev=False,
+    )
+
+    parser.add_argument(
+        "--usage",
+        help="Key usage",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaUsage],
+    )
+
+    parser.add_argument(
+        "--id",
+        help="Key identifier",
+        type=lambda number_string: int(number_string, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--type",
+        help="Key type",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaKeyType],
+    )
+
+    parser.add_argument(
+        "--size",
+        help="Key size in bits",
+        type=lambda number_string: int(number_string, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--algorithm",
+        help="Ket algorithm",
+        type=str,
+        required=False,
+        default="NONE",
+        choices=[x.name for x in PsaAlgorithm],
+    )
+
+    parser.add_argument(
+        "--location",
+        help="Storage location",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaKeyLifetime],
+    )
+
+    parser.add_argument(
+        "--key",
+        help="Key value",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--file",
+        help="JSON file to create or modify",
+        type=str,
+        required=False,
+    )
+
+    parser.add_argument(
+        "--cracen_usage",
+        help="CRACEN KMU Slot usage scheme",
+        type=str,
+        required=False,
+        default="NONE",
+        choices=[x.name for x in PsaCracenUsageSceme],
+    )
+
+    args = parser.parse_args()
+
+    metadata = binascii.hexlify(PlatformKeyAttributes(key_type=PsaKeyType[args.type],
+                              identifier=args.id,
+                              location=PsaKeyLifetime[args.location],
+                              usage=PsaUsage[args.usage],
+                              algorithm=PsaAlgorithm[args.algorithm],
+                              size=args.size,
+                              cracen_usage=PsaCracenUsageSceme[args.cracen_usage])
+                                .pack()).decode('utf-8').upper()
+
+    if args.file and Path(args.file).is_file():
+        with open(args.file, encoding="utf-8") as file:
+            json_data= json.load(file)
+    else:
+        json_data= json.loads('{ "version": 0, "keyslots": [ ]}')
+
+
+    if args.key != "TRNG":
+        key = args.key.strip("0x")
+        if not is_valid_hexa_code(key):
+            print("Invalid KEY value")
+            return
+        value = f'0x{key}'
+    else:
+        key = args.key
+        value  = f'{key}:{int(math.ceil(args.size / 8))}'
```

Comment:
What is the purpose of the ceil and division?

Reply:
The size input is in bits, for instance AES is 256 bit while ED25519 is 255 bit for some reason, but when I generate the key I must still generate a key size divisible by 8 (in bytes). So I need to generate 32 bytes of data for both AES and ED25519. So to make sure I get enough bytes, I need to make sure that I always round up to the closest number.